app/controllers/api/v1/users_controller.rb

```diff
@@ -1,11 +1,37 @@
 module Api
   module V1
     class UsersController < ApplicationController
-      before_action :authenticate_user!, only: [:targets]
+      before_action :authenticate_user!, only: %i[update targets]
+
+      def show; end
+
+      def update
+        user.update!(user_params)
+        render partial: 'show', locals: { user: current_user }, status: :ok
+      end
 
       def targets
```

Comment:
I would remove this action from this controller and use the same logic on the `index` action of the targets controller, where you should return only the targets of the current user instead of all the targets. This is not part of the PR but it is worth changing (remember to also remove the related route and view)